.github/workflows/rawls-run-azure-e2e-tests.yaml

```diff
@@ -83,14 +94,16 @@ jobs:
       contents: 'read'
       id-token: 'write'
     outputs:
-      custom-version-json: ${{ steps.render-rawls-version.outputs.custom-version-json }}
+      custom-version-json: ${{ inputs.build-branch && steps.render-rawls-version.outputs.custom-version-json || ''}}
     steps:
       - uses: 'actions/checkout@v3'
+        if: ${{ inputs.build-branch }}
```

Comment:
I tried to skip the whole job instead of each individual step, but that caused problems with handling dependencies of later jobs (I couldn't figure out how to make this job optional and still enforce "needs" when this job is required).

As it is, this job is very fast if no build is actually necessary, and it handles converting `custom-version-json` to the empty string in that case, which simplifies later jobs.

Reply:
Example output when we don't build: https://github.com/broadinstitute/rawls/actions/runs/7214169971/job/19655669762